src/mcts/search.cc

```diff
@@ -558,7 +562,7 @@ void SearchWorker::GatherMinibatch() {
     ++nodes_found;
     // If node is already known as terminal (win/loss/draw according to rules
     // of the game), it means that we already visited this node before.
-    if (node->IsTerminal()) continue;
+    if (node->IsTerminal() && node != search_->root_node_) continue;
```

Comment:
I don't think this is a proper fix for whatever its trying to fix. ExtendNode won't unmark the node from its previous terminal marked state, so while it'll have edges added there won't be a NN evaluation done (I think crem made the same comment in discord).
The tree reuse path instead should be changed to perform a trim tree at head, if the new head is terminal.

Reply:
I like that idea. That would also be useful to other non-tb analyses requested of root nodes as well (such as turning off 50 move rule and others)